### PR TITLE
Add a DateTime editor widget qml test

### PR DESCRIPTION
**Pull request by nirvn**

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 1c9a02fe137893a4b1dda39b9bcba82554ffc5f9: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3283-65f27ccffd9cb6bbdc832420a6b5f5d3b27f3082-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3283-65f27ccffd9cb6bbdc832420a6b5f5d3b27f3082-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3283-65f27ccffd9cb6bbdc832420a6b5f5d3b27f3082-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3283-65f27ccffd9cb6bbdc832420a6b5f5d3b27f3082-x86-android.apk)
</details>

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -52,16 +52,46 @@ macro (ADD_QFIELD_QML_TEST TESTNAME TESTSRC)
   get_target_property(QT_TARGET_TYPE ${QT_PKG}::Core TYPE)
   if(${QT_TARGET_TYPE} STREQUAL "STATIC_LIBRARY")
     include("${CMAKE_SOURCE_DIR}/cmake/QtQmlImportScanner.cmake")
+
     file(GLOB test_qmlfiles "${CMAKE_CURRENT_SOURCE_DIR}/qml/*.qml")
     file(GLOB qfield_qmlfiles "${CMAKE_SOURCE_DIR}/src/qml/*.qml")
+    file(GLOB qfield_qmlfiles_geometryeditors
```
Comment (m-kuhn) [inline test/CMakeLists.txt:58]: We need to bring this into qml/cmakelists.txt if we don't want to forget to update one or the other in the future.

Comment (nirvn) [reply test/CMakeLists.txt:58]: @m-kuhn , can I just relocate the whole code block into src/qml/CMakeLists.txt ?

Comment (nirvn) [reply test/CMakeLists.txt:58]: Ignore, I understand what you want of me sir ;-)

File: src/qml/CMakeLists.txt [hunk 2]
```diff
@@ -34,3 +34,30 @@ set_target_properties(qfield_qml PROPERTIES LINKER_LANGUAGE CXX)
...
+       "${CMAKE_SOURCE_DIR}/src/qml/imports/QFieldControls/+Qt6/*.qml")
+else()
+  file(GLOB qfield_qmlfiles_theme_extra
+       "${CMAKE_SOURCE_DIR}/src/qml/imports/Theme/+Qt5/*.qml")
+  file(GLOB qfield_qmlfiles_qfieldcontrols
+       "${CMAKE_SOURCE_DIR}/src/qml/imports/QFieldControls/+Qt5/*.qml")
+endif()
+list(
+  APPEND
+  qfield_qmlfiles
```
Comment (m-kuhn) [inline src/qml/CMakeLists.txt:58]: I think a `set(...PARENT_SCOPE...)` will be needed to make this var accessible from elsewhere.

Comment (nirvn) [reply src/qml/CMakeLists.txt:58]: @m-kuhn , TIL :bowing_man:

Comment (nirvn) [reply src/qml/CMakeLists.txt:59]: @m-kuhn , done.

File: src/qml/CMakeLists.txt [hunk 3]
```diff
@@ -63,6 +63,4 @@ list(
   ${qfield_qmlfiles_theme_extra}
   ${qfield_qmlfiles_qfieldcontrols})
 
-set(qfield_qmlfiles
-    ${qfield_qmlfiles}
-    PARENT_SCOPE)
+set_target_properties(qfield_qml PROPERTIES QML_FILES "${qfield_qmlfiles}")
```
Comment (m-kuhn) [inline src/qml/CMakeLists.txt:66]: `set_property(TARGET qfield_qml PROPERTY QML_FILES ${qfield_qmlfiles})` will preserve the list as list

Comment (nirvn) [reply src/qml/CMakeLists.txt]: @m-kuhn , done.

**Review by m-kuhn — APPROVED** [review]

**Comment by m-kuhn** [comment]
Windows fix confirmed

**Comment by nirvn** [comment]
iOS fix confirmed too using the very last commit (i.e. set_property(...)).

**Pull request merged**